src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -113,6 +113,17 @@ public Http2Stream(HttpRequestMessage request, Http2Connection connection, int s
                     _requestBodyCancellationSource = new CancellationTokenSource();
                 }
 
+                if (_request.HasHeaders && _request.Headers.ExpectContinue == true)
+                {
+                    // Create a TCS that will complete when one of two things occurs:
+                    // 1. if a timer fires before we receive the relevant response from the server.
+                    // 2. if we receive the relevant response from the server before a timer fires.
+                    // In the first case, we could run this continuation synchronously, but in the latter, we shouldn't,
+                    // as we could end up starting the body copy operation on the main event loop thread, which could
+                    // then starve the processing of other requests.  So, we make the TCS RunContinuationsAsynchronously.
+                    _expect100ContinueWaiter = new TaskCompletionSource<bool>(TaskContinuationOptions.RunContinuationsAsynchronously);
```

Comment:
Should we add some asserts here? For example, it is assumed that `_expect100ContinueWaiter` is `null` already? Or can we possibly create a new TCS and overwrite the existing one? If so, should be check the state of the existing one before doing that?

Reply:
Looking further in this PR, I see that we used to reset the `_expect100ContinueWaiter` to `null`. But now we're not doing that. Does it ever get reset to `null` anymore?

Reply:
This is the constructor, so it's guaranteed to be null here.